File: insertion_sort/typescript/insertion_sort.test.ts
import insertion_sort from './insertion_sort';

test('sorting an array', () => {
    const arr = Array.from({ length: 40 }, () => Math.floor(Math.random() * 40));
    const correctlySortedArr = [...arr]
    correctlySortedArr.sort((a, b) => a - b)

    insertion_sort(arr, arr.length)

    expect(arr).toEqual(correctlySortedArr);
});

File: insertion_sort/go/insertion_sort_test.go
package insertionsort_test

import (
	"slices"
	"testing"

	insertionsort "github.com/alirostami1/dsa/go/insertion_sort"
	"github.com/stretchr/testify/assert"
)

func TestInsertionSort(t *testing.T) {
	arr := []int{5, 2, 4, 3, 1}

	correctlySortedArr := make([]int, len(arr))
	copy(correctlySortedArr, arr)
	slices.Sort(correctlySortedArr)

	insertionsort.InsertionSort(arr, len(arr))

	assert.Equal(t, correctlySortedArr, arr)
}
